internal/proto/codec/builtin_test.go

```diff
@@ -586,10 +586,30 @@ func TestEntryListUUIDListIntegerCodec_Decode(t *testing.T) {
 
 	// then
 	assert.Equal(t, len(result), 1)
-	assert.Equal(t, result[0].Key.([]types.UUID)[0].String(), key.String())
+	assert.Equal(t, result[0].Key.(types.UUID).String(), key.String())
 	assert.EqualValues(t, result[0].Value.([]int32), value)
 }
 
+func TestEntryListUUIDListIntegerCodec_Decode_Many(t *testing.T) {
+	// given
+	clientMessage := proto.NewClientMessageForEncode()
+	entries := []proto.Pair{
+		{Key: types.NewUUID(), Value: []int32{1, 2, 3}},
+		{Key: types.NewUUID(), Value: []int32{4, 5, 6}},
+	}
+	EncodeEntryListUUIDListInteger(clientMessage, entries)
+
+	// when
+	result := DecodeEntryListUUIDListInteger(clientMessage.FrameIterator())
+
+	// then
+	assert.Equal(t, len(result), 2)
+	assert.Equal(t, result[0].Key.(types.UUID).String(), entries[0].Key.(types.UUID).String())
+	assert.EqualValues(t, result[0].Value.([]int32), entries[0].Value)
+	assert.Equal(t, result[1].Key.(types.UUID).String(), entries[1].Key.(types.UUID).String())
+	assert.EqualValues(t, result[1].Value.([]int32), entries[1].Value)
+}
```

Comment:
New test to cover the scenario of a multi-node cluster providing a map of partitions.

Reply:
Thanks for this test! Do you think we can merge it the test above this one? It would be better to have a single one instead of separate ones.

Reply:
My pleasure!

From what I've seen in the code-base, each test ensures a single behavior. Are you sure you want me to break that pattern and merge two distinct test cases into a single function? (I'm happy to follow any convention, just double checking)

Reply:
Oh you are right! I would say let's keep it as is. Resolving this